### PR TITLE
Add support for compression (gzip and deflate)

### PR DESCRIPTION
**Pull request by sumo-drosiek**
Add support for compression basing on logstash plugin

https://github.com/SumoLogic/logstash-output-sumologic/blob/master/lib/logstash/outputs/sumologic/compressor.rb

### CODE REVIEW

File: lib/fluent/plugin/out_sumologic.rb [hunk 1]
```diff
@@ -89,6 +124,10 @@ class Fluent::Plugin::Sumologic < Fluent::Plugin::Output
   # https://help.sumologic.com/Manage/Fields
   desc 'Fields string (eg "cluster=payment, service=credit_card") which is going to be added to every record.'
   config_param :custom_fields, :string, :default => nil
+  desc 'Compress payload'
+  config_param :compress, :bool, :default => false
+  desc 'Encoding method of compresssion (either gzip or deflate)'
+  config_param :compress_encoding, :string, :default => SumologicConnection::COMPRESS_DEFLATE
```
Comment (samjsong) [inline lib/fluent/plugin/out_sumologic.rb:130]: I'm not too familiar with these, what was our reason for defaulting to deflate vs gzip?

Comment (sumo-drosiek) [reply lib/fluent/plugin/out_sumologic.rb:133]: I would say consistency with logstash plugin:

https://github.com/SumoLogic/logstash-output-sumologic/blob/master/lib/logstash/outputs/sumologic/compressor.rb

Deflate seems to be more compact (shorter header) and faster.
Gzip includes additional headers and data validation, but uses deflate underneath

Comment (sumo-drosiek) [reply lib/fluent/plugin/out_sumologic.rb:133]: Also gzip looks to be preferred by HTTP society

Comment (sumo-drosiek) [reply lib/fluent/plugin/out_sumologic.rb]: After some research I changed it to gzip. We can always change it back before release

**Pull request merged**